Detectors/Base/include/DetectorsBase/TreeStream.h

```diff
@@ -0,0 +1,206 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// @brief Class for creating debug root trees with std::cout like intervace
+/// @author Marian Ivanov, marian.ivanov@cern.ch (original code in AliRoot)
+///         Ruben Shahoyan, ruben.shahoyan@cern.ch (porting to O2)
+
+#ifndef ALICEO2_TREESTREAM_H
+#define ALICEO2_TREESTREAM_H
+
+#include <TString.h>
+#include <TTree.h>
+#include <vector>
+
+class TBranch;
+class TClass;
+class TDataType;
+
+namespace o2
+{
+namespace Base
+{
+class TreeStream
```

Comment:
I think it would be good to put explanations just before the class declaration. Not sure doxygen would pick up the text from above otherwise?

Reply:
Is DetectorsBase the best place for such a utility class?